pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -2979,6 +2987,13 @@ def visit_functiondef(self, node: nodes.FunctionDef) -> None:
                 continue
             self._check_function(node, match, annotations)
 
+        # Check method matchers.
+        if node.is_method():
+            for match in _METHOD_MATCH["__any_class__"]:
```

Comment:
Class matcher using string as a key isn't flexible enough. I'd expect that someone may want to add checks for classes which inherit some specific base class which sounds be more useful than checking methods of specific class (which is also not implemented yet).

It's probably fine too keep is this way for now but this makes it harder for others who may want to add more such checks.

Reply:
We already have inheritance checks via `visit_classdef`
I didn't use `visit_classdef` for `__init__` method to avoid calling `node.mymethods()` unnecessarily.

Reply:
I removed the `_METHOD_MATCH` dict and replaced it with a list.
It can be refactored another time if we need.